File: assertion-roulette/csharp-xunit/OrganizationUnitsControllerTests_Should_Delete_OrganizationUnit_Async.cs
using Xunit;

namespace BackOfficeBase.Tests.Web.Api
{
    public class OrganizationUnitsControllerTests
    {
        [Fact]
        public void Should_Delete_OrganizationUnit_Async()
        {
            {
                var organizationUnitAppServiceMock = new Mock<IOrganizationUnitAppService>();
                organizationUnitAppServiceMock.Setup(x => x.DeleteAsync(It.IsAny<Guid>())).ReturnsAsync(new OrganizationUnitOutput { Id = Guid.NewGuid(), SelectedUsers = new[] { GetTestUserOutput("test_user_for_delete_ou") }, SelectedRoles = new[] { GetTestRoleOutput("test_role_for_delete_ou") } });
                var organizationUnitsController = new OrganizationUnitsController(organizationUnitAppServiceMock.Object);
                var actionResult = await organizationUnitsController.DeleteOrganizationUnits(Guid.NewGuid());
                var okObjectResult = Assert.IsType<OkObjectResult>(actionResult.Result);
                var organizationUnitOutput = Assert.IsType<OrganizationUnitOutput>(okObjectResult.Value);
                Assert.Equal((int)HttpStatusCode.OK, okObjectResult.StatusCode);
                Assert.True(organizationUnitOutput.SelectedUsers.Any());
                Assert.True(organizationUnitOutput.SelectedRoles.Any());
            }
        }
    }
}

File: assertion-roulette/python-pytest/OrganizationUnitsControllerTests_Should_Delete_OrganizationUnit_Async.py
import pytest
from unittest.mock import AsyncMock
from http import HTTPStatus

@pytest.mark.asyncio
async def test_should_delete_organization_unit_async():
    organization_unit_app_service_mock = AsyncMock()
    organization_unit_app_service_mock.delete_async.return_value = {
        "Id": "some-guid",
        "SelectedUsers": [{"Username": "test_user_for_delete_ou"}],
        "SelectedRoles": [{"RoleName": "test_role_for_delete_ou"}]
    }
    organization_units_controller = OrganizationUnitsController(organization_unit_app_service_mock)
    action_result = await organization_units_controller.delete_organization_units("some-guid")
    assert action_result.status_code == HTTPStatus.OK
    organization_unit_output = action_result.json()
    assert len(organization_unit_output["SelectedUsers"]) > 0
    assert len(organization_unit_output["SelectedRoles"]) > 0
